### PR TITLE
fix(seeding): persist live OAG audits + quiet transient health-check noise

### LINKED ISSUE #121: Data pipeline failed: Validation — 2026-07-09
## Automated Data Pipeline Failure

**Failed Steps:** Validation
**Health Check:** ✅ All sources reachable
**Run:** https://github.com/Rodgers31/audit_app/actions/runs/28991458540
**Triggered by:** schedule
**Domain:** all

### Action Required
1. Check the [workflow run logs](https://github.com/Rodgers31/audit_app/actions/runs/28991458540)
2. Verify the `ingestion_jobs` table for domain-level errors
3. If a data source is down, the workflow will retry tomorrow automatically
4. For persistent failures, check if the upstream API/URL has changed

### PR DESCRIPTION
**Pull request by Rodgers31**
## Root cause

The nightly **"Data pipeline failed: Validation"** issues (#121–132) all share one cause. Since PR #120 (the slow-PDF fix) let seeding pass, the pipeline advanced to the validation gate — which fails every night on:

```
[FAIL] Audit Records: 25 rows (expected >= 50)
[FAIL] Audits with audit_year: 25 rows (expected >= 50)
[FAIL] Audits with query_type: 25 rows (expected >= 30)
```

The audits domain **fetches real OAG findings every night but silently drops 100% of them.** From a live run: `Successfully fetched audit data from OAG (14 findings)` → `[OK] audits: completed | created=0 updated=0 processed=0`.

The fetcher emitted each finding with `period_label=""` and **no** `start_date`/`end_date`/`audit_year` (`"Will be extracted from context"` — never implemented). The parser drops any finding missing those fields, so nothing persisted and the audits table stayed frozen at the 25 real records kept after the 2026-07-07 fabricated-data purge — below the stale `>= 50` threshold (calibrated against the old, partly-fabricated 537-row volume).

## Fixes

1. **Fetcher — derive the fiscal period** (`period_label`, July→June `start_date`/`end_date`, end-year `audit_year`) from the OAG report filename first (`…-Homa-Bay-2021-2022.pdf`, `…-NATIONAL-GOVERNMENT-2024-2025.pdf`), then the report text (`for the year ended 30 June 2022`). `finditer` scans past the `/uploads/2023/11/` publish date in the URL; a consecutive-year guard rejects multi-year ranges. Findings now survive the parser and persist.
2. **Fetch↔parse guard** — the audits domain now WARNs when findings are fetched but the parser keeps fewer (or zero), so a shape regression can't hide for days behind `processed=0`.
3. **Recalibrated validation thresholds** — audit gates are now regression *floors* (20) matching the honest post-purge baseline (25), not the stale `>= 50`. Pipeline health (did ingestion persist?) is tracked by the ingestion-job counts + the new WARN, not by inflating the floor.
4. **Health-check noise (#124)** — blocked sources are re-probed with a browser UA; WAF/transient codes (`403/415/429/000/5xx`) are classified WARNING instead of CRITICAL so an intermittent WAF block on the GitHub Actions runner IP stops filing issues; open `data-source-down` issues auto-close on a green run. Genuine `404/410` still honour the per-source critical flag so real URL migrations still page.

## Tests

`backend/tests/test_audit_fetcher_period.py` — proves fetcher output now survives the parser (the exact regression), plus fiscal-period derivation edge cases (2-/4-digit spans, upload-path date, `year ended` fallback, multi-year rejection). `104 passed` across the audit/seeding/validation unit files.

## Notes

- The nightly only stops failing once this reaches `main` (runs 02:00 UTC). A run before merge can still file one more duplicate.
- Extraction *quality* of the naive regex parser is unchanged and out of scope here; this PR fixes the persistence contract so real findings land.

Closes #121, #122, #123, #125, #126, #127, #128, #129, #130, #131, #132

### CODE REVIEW

**Comment by Rodgers31** [comment]
Thanks @copilot — all four addressed in 6c47baf. Verified each against actual behaviour first:

**1. `probe()` can emit `000000` (seed.yml) — valid, fixed.** Reproduced it: a dead host yields `000000` (curl prints `000` to stdout *and* exits non-zero, then `|| echo "000"` appends a second `000`). That defeats `is_transient` (which matches `000` exactly), so a genuine outage of a *critical* source would wrongly escalate to CRITICAL — and it matches the real `000000` shown for Treasury in #124. `probe()` now captures stdout, guards the assignment against `set -e` (GitHub runs `run:` blocks with `-e`) via `|| true`, and normalises anything that isn't a bare 3-digit code to `000`. Verified under `set -e` against dead (`000`, transient) and live (`200`) hosts.

**2 & 3. WARN wording (audits `__init__.py`) — valid, fixed.** The parser drops on any of `entity_slug/entity/severity/finding_text` **and** `period_label/start_date/end_date`. Both messages now name the full contract instead of implying only the fiscal-period fields (while still flagging those as the usual suspect).

**4. Auto-close closes all `data-source-down` issues — valid, fixed.** Now requires **both** `data-source-down` *and* `automated` labels (comma = AND), which every workflow-generated outage issue carries — so a human-filed investigative issue using just `data-source-down` is never auto-closed.

Tests still green (`50 passed` across the audit fetcher/parser files; YAML validated).

File: .github/workflows/seed.yml [hunk 1]
```diff
@@ -113,26 +113,67 @@ jobs:
...
-              "$url" 2>/dev/null || echo "000")
+            HTTP_CODE=$(probe "$PRIMARY_UA" "$url")
+            # On a block-style code, re-probe once with a browser UA before
+            # concluding the source is unhealthy.
+            if is_transient "$HTTP_CODE"; then
+              RETRY_CODE=$(probe "$BROWSER_UA" "$url")
+              if [[ "$RETRY_CODE" =~ ^[23] ]]; then
+                HTTP_CODE="$RETRY_CODE"
+              fi
+            fi
```
[the remark this answers is not in the document]
Comment (Rodgers31) [reply .github/workflows/seed.yml:174]: Good catch — valid, fixed in 9487e6a. The browser-UA re-probe now adopts the retry code whenever it is **definitive** (not itself transient), rather than only on 2xx/3xx: a 404/410 revealed past the WAF now falls through to the critical-flag branch instead of being masked behind the primary transient code. Kept the original code only when the re-probe is *also* transient (block is real/IP-based). Verified matrix: `415→200`=OK, `415→404/410`=CRITICAL, `415→415` & `000→000`=WARNING(transient), `503→301`=OK.
